### PR TITLE
python3Packages.robotframework-tidy: init at 4.11.0

### PR DESCRIPTION
**Pull request by otavio**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/robotframework-tidy/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "MarketSquare";
+    repo = "robotframework-tidy";
+    rev = "${version}";
+    hash = "sha256-pWW7Ex184WgnPfqHg5qQjfE+9UPvCmE5pwkY8jrp9bI=";
+  };
+
+  nativeBuildInputs = [
```
Comment (NickCao) [inline pkgs/development/python-modules/robotframework-tidy/default.nix:26]:
```suggestion
  build-system = [
```

File: pkgs/development/python-modules/robotframework-tidy/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
```
Comment (NickCao) [inline pkgs/development/python-modules/robotframework-tidy/default.nix:30]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/robotframework-tidy/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    robotframework
+    click
+    colorama
+    pathspec
+    tomli
+    rich-click
+    jinja2
+  ];
+
+  meta = with lib;   {
```
Comment (NickCao) [inline pkgs/development/python-modules/robotframework-tidy/default.nix:40]: Please format with `nixfmt-rfc-style`

**Comment by otavio** [comment]
@NickCao I applied all fixes. Thanks for reviewing.

**Review by NickCao — APPROVED** [review]
diff LGTM.
Result of `nixpkgs-review pr 298365` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.robotframework-tidy</li>
    <li>python311Packages.robotframework-tidy.dist</li>
    <li>python312Packages.robotframework-tidy</li>
    <li>python312Packages.robotframework-tidy.dist</li>
  </ul>
</details>

**Review by dotlambda — COMMENTED** [review]
Why put this under `python3Packages`? It doesn't seem to have an API.
Also, `meta.mainProgram` and `nativeCheckInputs` are missing.

**Pull request merged**